src/main/java/com/icthh/xm/ms/entity/service/impl/FunctionServiceImpl.java

```diff
@@ -53,18 +60,15 @@ public FunctionContext execute(final String functionKey, final Map<String, Objec
         Objects.requireNonNull(functionKey, "functionKey can't be null");
         Map<String, Object> vInput = CustomCollectionUtils.emptyIfNull(functionInput);
 
-        FunctionSpec functionSpec = xmEntitySpecService.findFunction(functionKey).orElseThrow(
-            () -> new IllegalArgumentException("Function not found, function key: " + functionKey));
+        dynamicPermissionCheckService.checkContextPermission(DynamicPermissionCheckService.FeatureContext.FUNCTION,
+            "FUNCTION.CALL", functionKey);
```

Comment:
Maybe better move in constant or in the configuration?

Reply:
if You mean  privilege name "FUNCTION.CALL", yes. I will do it on phase 2, now this code needed to visualize integration problem with FUNCTION.CALL.GET, FUNCTION.CALL.PUT permissions

Reply:
Fixed